modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/redownload/AlignmentCheck.scala

```diff
@@ -0,0 +1,89 @@
+package org.tessellation.sdk.infrastructure.redownload
+
+import cats.MonadThrow
+import cats.data.NonEmptySet
+import cats.effect.kernel.Temporal
+import cats.effect.std.Supervisor
+import cats.effect.syntax.concurrent._
+import cats.effect.{Async, Ref}
+import cats.syntax.applicative._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.set._
+
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+import org.tessellation.ext.collection.IterableUtils.pickMajority
+import org.tessellation.schema.peer.{L0Peer, Peer, PeerInfo}
+import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshotInfo}
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.cluster.storage.{ClusterStorage, L0ClusterStorage}
+import org.tessellation.sdk.domain.seedlist.SeedlistEntry
+import org.tessellation.sdk.domain.snapshot.storage.LastSnapshotStorage
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+
+trait AlignmentCheck[F[_]] {
+  def check(): F[Unit]
+}
+
+object AlignmentCheck {
+  private val maxConcurrentPeerInquiries = 10
+  def make[F[_]: Async: Temporal](
+    priorityPeers: NonEmptySet[SeedlistEntry],
+    isAlignedR: Ref[F, Boolean],
+    clusterStorage: ClusterStorage[F],
+    l0ClusterStorage: L0ClusterStorage[F],
+    lastSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+    l0GlobalSnapshot: L0GlobalSnapshotClient[F],
+    waitForPeers: FiniteDuration = 1.minute
+  ): AlignmentCheck[F] = () => {
+    // TODO Are these the correct peers to use to determine if we're aligned?
+    //  At first I thought to use the priority seed list but I'm not sure that's correct
+    def getPeers: F[List[Peer]] =
+      priorityPeers.toNonEmptyList
+        .traverse(se => clusterStorage.getPeer(se.peerId))
```

Comment:
I think you are using a wrong storage here. I guess you meant to use `l0ClusterStorage`. After you use it you will already get back `L0Peer`'s. I guess the peer's type indicates that you are using the wrong one (`Peer` implicitly means the same layer node).

Reply:
Just to summarize point of discussion related to this point from meeting: we don't need to update / reset the peers in l0ClusterStorage if we're relying on user to provide initial join peer and assuming priority peers will be a part of this. Since user supplies the initial through environment, we're assuming the priority embedded default list will be a part of that (unless overridden.) 

Again also, since we're making sure each downloaded snapshot (GL0 specific) conforms to that priority list -- then we also don't need to deal with the update process since we're guaranteed that that peer list will still contain our priority peers as it has already undergone the check.

For the more complex situations -- i.e. where those peers are not already contained, we'll need a separate solution (involving persisting the IPs within priority seed list ala the new fields chuong added,) where we modify the peers dynamically by querying by having the IP information embedded. For now, we do not need to guarantee this as this will address most situations presently, (and still has the fallback of user adjusting the GL0 peer they're joining against initially.)

Reply:
Resolving; code no longer present